autoit_unpack.py

```diff
@@ -1,18 +1,41 @@
-from typing import Optional, Tuple
+import logging
 import struct
+from enum import Enum
+from typing import Optional, Tuple
+
+import lief  # type: ignore
+
 from opcodes import OPCODES
-import logging
-from utils import BitStream, decrypt_lame, crc_data
+from utils import BitStream, crc_data, decrypt_lame, decrypt_mt, filetime_to_dt
 
-# 10 megabytes
-MAX_SCRIPT_SIZE = 10 * 10**6
+EA05_MAGIC = bytes.fromhex("a3484bbe986c4aa9994c530a86d6487d41553321")
+
+
+class AutoItVersion(Enum):
+    EA05 = 0
+    EA06 = 1
 
 
+# 10 megabytes
+MAX_SCRIPT_SIZE = 10 * 10 ** 6
+
 log = logging.getLogger(__name__)
 
 
-def decompress_ea06(data: bytes) -> Optional[bytes]:
-    if data[:4] != b"EA06":
+def get_script_resource(pe: lief.PE) -> Optional[lief.PE.ResourceDirectory]:
+    for child in pe.resources.childs:
+        for grandchild in child.childs:
+            if grandchild.has_name and grandchild.name == "SCRIPT":
+                return grandchild
+    return None
+
+
+def decompress(data: bytes, version: AutoItVersion) -> Optional[bytes]:
+    if data[:4] == b"EA05" and version == AutoItVersion.EA05:
+        pass
+    elif data[:4] == b"EA06" and version == AutoItVersion.EA06:
+        pass
+    else:
         log.error("Magic mismtach")
```

Comment:
Maybe log magic just in case? It'll be much easier to debug in case it actually happens :thinking: 

(Also, a typo)

```suggestion
        log.error("Magic mismatch: %s", data[:4])
```